content/kubermatic/master/upgrading/2.14_to_2.15/_index.en.md

```diff
@@ -5,7 +5,384 @@ weight = 90
+Using this command is recommended, as it does all conversions in a single step, because the old
+`values.yaml` contained all information in some form or another. Make sure to add `--pause-seeds`
+to automatically add the `operator.kubermatic.io/skip-reconciling` annotation.
+
+```bash
+./kubermatic-installer convert-helm-values --pause-seeds myvalues.yaml
+# apiVersion: operator.kubermatic.io/v1alpha1
+# kind: KubermaticConfiguration
+# metadata:
+#   name: kubermatic
+#   namespace: kubermatic
+# ...
+#
+# ---
+# apiVersion: kubermatic.k8s.io/v1
+# kind: Seed
+# ...
+#
+# ---
+# ...
+```
+
+In case migrating the Helm values.yaml is not feasable, the installer also offers a dedicated
+command to convert just the `datacenters.yaml` into Seeds/Secrets. By default, the `datacenters.yaml`
+did not contain kubeconfigs, so you have to manually provide a kubeconfig with contexts for every
+seed cluster in the `datacenters.yaml` (the installer will error out if contexts are missing).
+If you do not provide a kubeconfig, Seeds will still be converted, but you are responsible
+yourself to provide the appropriate kubeconfig Secrets and to reference them in the Seeds.
+
+```bash
+./kubermatic-installer convert-datacenters --pause-seeds --kubeconfig kubeconfig-with-all-seeds datacenters.yaml
+# apiVersion: kubermatic.k8s.io/v1
+# kind: Seed
+# ...
+#
+# ---
+# ...
+```
+
+Refer to the [KubermaticConfiguration]({{< ref "../../concepts/kubermaticconfiguration" >}}) and
+[Seed]({{< ref "../../concepts/seeds" >}}) documentation for more information.
+
+Once the conversion is completed, carefully check the new configuration files for mistakes. Do
+note that the installer generally does not output default values, so if you configured for example
+2 Apiserver replicas and this is the default for Kubermatic anyway, the generated `KubermaticConfiguration`
+will skip the value alltogether, relying on the defaulting.
+
+It is now recommended to setup Kubermatic in a test environment, using the newly converted files.
+
+#### Upgrading the Master Cluster
+
+It's now time to change the Helm charts. Begin by uninstalling the `kubermatic` chart. This will make
+the dashboard unavailable and begin the downtime.
+
+**Helm 3**
+
+```bash
+helm --namespace kubermatic delete kubermatic
+```
+
+**Helm 2**
+
+```bash
+helm --tiller-namespace kubermatic-installer delete --purge kubermatic
+```
+
+Once the Helm chart is gone, it's time to perform a clean installation of the new Kubermatic Operator.
+The first step for that is to add new Kubermatic CRDs:
+
+```bash
+kubectl apply -f charts/kubematic/crd/
+```
+
+To prevent Kubermatic from rejecting existing clusters, make sure to apply the Seeds before installing
+the Operator. We assume that `seeds-with-secrets.yaml` is part of the output of one of the conversion
+commands described above.
+
+```bash
+kubectl apply -f seeds-with-secrets.yaml
```

Comment:
Is there any particular reason why you want to apply `seeds`, `secrets` but not `kubermaticconfig`?
As the tool outputs all manifests together it would be more practical to apply them all before the operator chart. e.g . `./kubermatic-installer convert-helm-values --pause-seeds myvalues.yaml | k apply -f -`

Reply:
I was thinking the same, but then I already had it written this way and just left it. I presume that uses will want to review the output from the conversion commands anyway before applying, but we can add an example showing that you _can_ just pipe it into kubectl if you're #yolo enough ^^

Reply:
Once the resources are applied they are not taken into account until the kubermatic operator is deployed, the users still have the opportunity of reviewing them using `kubectl`